examples/llm-vicuna/README.md

```diff
@@ -0,0 +1,88 @@
+
+* [Blog post](https://vicuna.lmsys.org/)
+* [Demo](https://chat.lmsys.org/)
+* [Repo](https://github.com/lm-sys/FastChat)
+
+## Prerequisites
+Install the latest SkyPilot and check your setup of the cloud credentials:
+```bash
+pip install git+https://github.com/skypilot-org/skypilot.git
+sky check
+```
+Find the Vicuna finetuning [SkyPilot YAMLs](examples/llm-vicuna/finetune.yaml)
+Find the Vicuna serving [SkyPilot YAMLs](examples/llm-vicuna/serve.yaml)
+
+## Serve the official Vicuna model by yourself with SkyPilot
+
+1. Start the serving the Vicuna-7B model on a single A100 GPU:
+```bash
+sky launch -c vicuna-serve -s serve.yaml
+```
+2. Check the output of the command. There will be a sharable gradio link (like the following). Open it in your browser to chat with Vicuna.
+```
+(task, pid=20933) 2023-04-12 22:08:49 | INFO | gradio_web_server | Namespace(host='0.0.0.0', port=None, controller_url='http://localhost:21001', concurrency_count=10, model_list_mode='once', share=True, moderate=False)
+(task, pid=20933) 2023-04-12 22:08:49 | INFO | stdout | Running on local URL:  http://0.0.0.0:7860
+(task, pid=20933) 2023-04-12 22:08:51 | INFO | stdout | Running on public URL: https://<random-hash>.gradio.live
+```
+
+3. [Optinoal] Try other GPUs:
+```bash
+sky launch -c vicuna-serve-v100 -s serve.yaml --gpus V100
+```
+
+4. [Optional] Serve the 13B model instead of the default 7B:
+```bash
+sky launch -c vicuna-serve -s serve.yaml --env MODEL_SIZE=13
+```
+
+
+## Finetuning Vicuna with SkyPilot
+Currently, training requires GPUs with 80GB memory.  See `sky show-gpus --all` for supported GPUs.
+
+We can start the finetuning of Vicuna model on the dummy data [dummy.json](examples/llm-vicuna/dummy-data) **with a single command**. It will automatically find the available cheapest VM on any cloud.
+
+To finetune on your own data, replace the file with your own, or change the line `/data/mydata.json: ./dummy.json` to the path of your own data.
+
+1. Replace the bucket name in [finetune.yaml](examples/llm-vicuna/finetune.yaml) with some unique name, so the SkyPilot can create a bucket for you to store the model weights. See `# Change to your own bucket` in the YAML file.
+
+2. **Finetune the Vicuna-7B model on 8 A100 GPUs (80GB memory) using spot instances**:
+```bash
+# Launch it on managed spot to save 3x cost
+sky spot launch -n vicuna scripts/train-vicuna.yaml --env WANDB_API_KEY
+```
+
+[Optional] Finetune a larger model 13B model
+```
+# Train a 13B model instead of the default 7B
+sky spot launch -n vicuna-7b scripts/train-vicuna.yaml --env WANDB_API_KEY --env MODEL_SIZE=13
+
+# Use *unmanaged* spot instances (i.e., preemptions won't get auto-recovered).
+# Unmanaged spot saves the cost of a small controller VM.  We recommend using managed spot as above.
+sky launch -n vicuna scripts/train-vicuna.yaml --env WANDB_API_KEY
```

Comment:
should this be `finetune.yaml`? same for other commands

maybe change to `WANDB_API_KEY=offline` too?

Reply:
Ahh good catch! Fixed. Also, turned wandb off by default.